MedallionShell/Shell.cs

```diff
@@ -76,6 +83,24 @@ public Command Run(string executable, IEnumerable<object>? arguments = null, Act
             return command;
         }
 
+        // internal for testing
+        internal static string? GetFullPathUsingSystemPathOrDefault(string executable)
+        {
+            var paths = Environment.GetEnvironmentVariable("PATH")!.Split(Path.PathSeparator);
+
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            {
+                var pathExtensions = Environment.GetEnvironmentVariable("PATHEXT")!
+                    .Split(Path.PathSeparator)
+                    .Concat(new[] { string.Empty })
+                    .ToArray();
+                return paths.SelectMany(path => pathExtensions.Select(pathExtension => Path.Combine(path, executable + pathExtension)))
+                    .FirstOrDefault(File.Exists);
+            }
+
+            return paths.Select(path => Path.Combine(path, executable)).FirstOrDefault(File.Exists);
```

Comment:
By checking `File.Exists`, that means we'll skip a _directory_ that matches. Is that the OS behavior?

For example, I type `dotnet` and there is a directory `dotnet` or `dotnet.exe`. What does the OS do?